integration/simulation/simulation.go

```diff
@@ -3,183 +3,99 @@ package simulation
+		l2NodeConfig:     &l2NodeCfg,
+		l2Network:        l2NetworkCfg,
+		avgBlockDuration: avgBlockDuration,
 	}
-
-	timeInUs := simulationTime * 1000 * 1000
-	go injectUserTxs(wallets, &l1Network, &l2Network, avgBlockDuration, timeInUs, &stats)
-
-	// Wait for the simulation time
-	time.Sleep(common.Duration(uint64(timeInUs)))
-
-	fmt.Println("Stopping..")
-
-	// stop L2 first and then L1
-	go l2Network.Stop()
-	go l1Network.Stop()
-
-	time.Sleep(time.Second)
-
-	return l1Network, l2Network
 }
 
-const INITIAL_BALANCE = 5000 // nolint:revive,stylecheck
-
-func injectUserTxs(wallets []wallet_mock.Wallet, l1Network ethereum_mock.L1Network, l2Network obscuro_node.L2Network, avgBlockDuration uint64, simulationTime int, s *Stats) {
-	// deposit some initial amount into every user
-	initialiseWallets(wallets, l1Network, avgBlockDuration, s)
+// RunSimulation executes the simulation given all the params
+// todo - introduce 2 parameters for nrNodes and random L1-L2 allocation
+// todo - random add or remove l1 or l2 nodes - logic for catching up
+func (s *Simulation) Start(
+	txManager *TransactionManager,
+	simulationTime int,
+) {
+	// todo - add observer nodes
+	// todo read balance
 
-	// inject numbers of transactions proportional to the simulation time, such that they can be processed
-	go injectRandomDeposits(wallets, l1Network, avgBlockDuration, simulationTime, s)
-	go injectRandomWithdrawals(wallets, l2Network, avgBlockDuration, simulationTime, s)
-	injectRandomTransfers(wallets, l2Network, avgBlockDuration, simulationTime, s)
-}
+	log.Log(fmt.Sprintf("Genesis block: b_%d.", common.ShortHash(common.GenesisBlock.Hash())))
 
-func initialiseWallets(wallets []wallet_mock.Wallet, l1Network ethereum_mock.L1Network, avgBlockDuration uint64, s *Stats) {
-	for _, u := range wallets {
-		tx := deposit(u, INITIAL_BALANCE)
-		t, _ := common.EncodeTx(tx)
-		l1Network.BroadcastTx(t)
-		s.Deposit(INITIAL_BALANCE)
-		time.Sleep(common.Duration(avgBlockDuration / 3))
-	}
-}
+	// todo - changing from time to common will delay the node start and it will not catch the first few blocks
+	s.l1Network.Start(time.Duration(s.avgBlockDuration / 4))
+	s.l2Network.Start(time.Duration(s.avgBlockDuration / 4))
 
-func injectRandomTransfers(wallets []wallet_mock.Wallet, l2Network obscuro_node.L2Network, avgBlockDuration uint64, simulationTime int, s *Stats) {
-	n := uint64(simulationTime) / avgBlockDuration
-	i := uint64(0)
-	for {
-		if i == n {
-			break
-		}
-		fromWallet := rndWallet(wallets)
-		from := fromWallet.Address
-		to := rndWallet(wallets).Address
-		if from == to {
-			continue
-		}
-		tx := wallet_mock.NewL2Transfer(from, to, common.RndBtw(1, 500))
-		signedTx := wallet_mock.SignTx(tx, fromWallet.Key.PrivateKey)
-		encryptedTx := enclave.EncryptTx(signedTx)
-		s.Transfer()
-		l2Network.BroadcastTx(encryptedTx)
-		time.Sleep(common.Duration(common.RndBtw(avgBlockDuration/4, avgBlockDuration)))
-		i++
-	}
-}
+	timeInUs := simulationTime * 1000 * 1000
```

Comment:
@otherview Is there a better name for this? What does `timeInUs` mean?

Reply:
Adding a comment to make it clearer.

Would be cool if we had some standard on time, like always using milli seconds? I feel like when we're going to Micros and nanos it's more error-prone.